### ISSUE TITLE
Issue pulling from reMarkable to Zotero with Nextcloud as WebDAV

### ISSUE THREAD

**Issue by scheuerbauer** [opal06/zotero2remarkable_bridge#10]
Apologies for my many issues - but I really really want to get this running.
Thx for the prior help. Figured, I'll open a new thread since it became a new issue :)

--
Here's the issue:
Successfully pushed a document from my Zotero which runs Nextcloud as WebDAV following these instructions (https://mlohr.com/sync-zotero-library-to-nextcloud/). All files are perfectly synched on the WebDAV, your app could perfectly pull it from the WebDAV and push it to ReMarkable, all great. Then I added some notes to the doc and moved it to my 'READ' folder, then I ran the "pull"-command and got this output:

![grafik](https://github.com/user-attachments/assets/3b3c4621-7ec9-4876-9705-2d0c6e3aeec9)

From what I understand, it failed to download the file (which it successfully reached at my RM) to the folder in my "TEMP" folder - since the specified folder does not exist any more. So I went to my TEMP folder and confirmed. Now I am curious, why that is, if that is a WebDAV issue from nextcloud? Any ideas?

**Comment by opal06** [comment]
I don't think it has something to do with Nextcloud or WebDAV. Rather, it seems like my dirty implementation of temp files bites me here. Since this program was mainly put together for my own needs, it was only used and tested on Linux. And while Python's pathlib library seems to have helped a lot in making the program run with Windows style path conventions, I haven't chosen the most robust way of getting the tempfile location where the program tries to store files it gets from remarkable's cloud for further processing. I will tty to fix this, but it won't happen tonight.

**Comment by scheuerbauer** [comment]
Any chance you're getting at that problem within the next 1-2 months?

**Comment by opal06** [comment]
Turns out it was actually not an issue with the temp path, but with rmapi having changed how its `get` function works, breaking the processing of the pdf. This should be fixed with this commit 8e2dc5d.

**Comment by scheuerbauer** [comment]
![grafik](https://github.com/user-attachments/assets/510246a0-19c2-465a-9cea-9ac3a2a0c8ed)

Almost...I updated with your last commit, then pushed file to remarkable, pulled it into the "read" folder and ran the "pull" command - this is the output....

**Comment by opal06** [comment]
I encountered this issue as I was trying to fix the other bug and it seems to come down to how Windows handels some special characters in the file names that rmapi puts out. One way to fix this would be to sanitise file names to not use special characters beyond maybe a minimal set like dashes. Unfortunately, I can't do this programmatically right now since the file name is the source of truth for matching the downloaded files back with their Zotero entries. Now I'm sure there are ways to handle this better, but I fear those might be beyond my skills right now. Apart from that, it seems like the git bash shell does a whole lot of funny things to paths that I can't really account for. 
Now, I think the best way to get over these issues would be to run the program in a true Linux environment. Don't worry, you don't need to switch to Linux or install it on your PC directly since Windows offers the Windows Subsystem for Linux which installs a minimal Ubuntu environment and gives you access to a "true" bash shell under Linux right inside Windows.

To install it, open a Windows PowerShell terminal and type 
```powershell
wsl --install
``` 
Once it finished downloading Ubuntu, it will prompt you for a restart. After the restart, you should see a terminal finishing the install. It will prompt you for a username and password and drop you into your Ubuntu bash shell. From here, I'd suggest updating Ubuntu's packages with
```bash
sudo apt update
sudo apt upgrade -y
```
Then just setup rmapi and zotero2remarkable following the respective Readmes. Note that `git` should already be installed, but you might need to install `go` and `python3-pip` by running:
```bash
sudo apt install golang-go python3-pip -y
```
After installing `rmapi` you also need to add go's install path to your $PATH variable to make it available:
```bash
echo "export PATH=\$PATH:/home/$USER/go/bin" >> ~/.bashrc
source ~/.bashrc
```
In case Ubuntu complains about the environment being externally managed during `pip` requirements install append `--break-system-packages` to the command and run again. It will be fine since we don't plan to run much else in this environment.

WSL mounts your Windows file system under `/mnt`, so to get your config file, run this command, but change your Windows user name accordingly:
```bash
 cp /mnt/c/Users/<Windows_User>/zotero2remarkable_bridge/config.yml ~/zotero2remarkable_bridge/
```
Ubuntu will also appear as an entry in your Windows start menu for future access.

**Comment by scheuerbauer** [comment]
Hey, thx a lot for the assistance. So with the ubuntu shell it works well so far, but still I have a download issue - a different one however, and as far as I understand it, it seems to be one of naming the temporary files? I think the program expects "..._remarks.pdf", but I only see "...-annotations.pdf" could that be the problem?

appreciate your help

![grafik](https://github.com/user-attachments/assets/8e8c0e4f-fd58-44cf-a72f-1ed41537252e)

**Comment by opal06** [comment]
This one is strange... The *-annotations.pdf file is misleading, since it's a by-product of the change to rmapi's `geta` function talked about above and not actually used in any way by the program since it is still experimental. Instead the program has been using the [remarks](https://github.com/lucasrla/remarks) library to generate the highlights and annotations. These are saved into the _remarks.pdf file. In the next step, this file is renamed to remove this appendix again and this is where it seems to fail. For me this indicates that remarks never ran/failed silently in some way and never writes the new pdf. Since remarks doesn't work on annotations created with remarkable OS versions >=3.0, which version are you running?

**Comment by scheuerbauer** [comment]
Hey there, thx for reopening. Weird..I am running a remarkable1 with rmOS 3.15.4.2...rmapi uses syncversion1.5...

**Comment by scheuerbauer** [comment]
Hey, 
ok, I think I got it working - up to the point that it is a remarks-issue - I guess. Could you quickly confirm, that the following issue is a [remarks](https://github.com/lucasrla/remarks) issue and not a zotero2rm-bridge issue? Then I could head over there and annoy lucasrla with my wishes...

![grafik](https://github.com/user-attachments/assets/dc0a4c4e-62bf-4dbf-b846-c3fa46e83378)

**Comment by scheuerbauer** [comment]
Oh I just saw that there might be a remarks fork that actually works with vm6:
https://github.com/lucasrla/remarks/issues/58#issuecomment-2156148243

is this a workeable branch?
https://github.com/Azeirah/remarks

**Comment by Azeirah** [comment]
> Oh I just saw that there might be a remarks fork that actually works with vm6: [lucasrla/remarks#58 (comment)](https://github.com/lucasrla/remarks/issues/58#issuecomment-2156148243)
> 
> is this a workeable branch? https://github.com/Azeirah/remarks

Hi!

I saw that you reached out to me via e-mail.

The branch I maintain for remarks indeed supports v3 and the RMPP. It powers the sync for https://scrybble.ink and handles hundreds of renders a day.

I haven't looked at the code for this project, but I assume it would be relatively trivial to switch from lucaslra's branch to mine.

**Comment by Potjoe-97** [comment]
> After installing `rmapi` you also need to ....

Hi, and thank you for your work. I'm following because I encountered the same issue in Windows and tried through WSL. Does this method still works for you ? I have an issue withrmapi in WSL (and only in WSL), where it seems to fail reaching sync servers after initial setup. I tried in a Debian and Ubuntu environment. 

` ERROR: 2025/05/19 09:37:45 main.go:79: failed to build documents tree, last error: failed to mirror was not ok: request failed with status 404`

**Comment by Azeirah** [comment]
> > After installing `rmapi` you also need to ....
> 
> Hi, and thank you for your work. I'm following because I encountered the same issue in Windows and tried through WSL. Does this method still works for you ? I have an issue withrmapi in WSL (and only in WSL), where it seems to fail reaching sync servers after initial setup. I tried in a Debian and Ubuntu environment.
> 
> ` ERROR: 2025/05/19 09:37:45 main.go:79: failed to build documents tree, last error: failed to mirror was not ok: request failed with status 404`

If you're having issues with rmapi, you should report it within the rmapi repository.

https://github.com/ddvk/rmapi

Just to confirm, do you have the latest version of rmapi (v0.0.30)?

**Comment by Potjoe-97** [comment]
> Just to confirm, do you have the latest version of rmapi (v0.0.30)?

Yes, and I tried both the last asset released and to build it myself. 

I asked here because it's the only place where running rmapi in WSL is discussed, in response to the original (and persisting) issue in windows. Rmapi is working fine in windows !

**Comment by Azeirah** [comment]
> > Just to confirm, do you have the latest version of rmapi (v0.0.30)?
> 
> Yes, and I tried both the last asset released and to build it myself.
> 
> I asked here because it's the only place where running rmapi in WSL is discussed, in response to the original (and persisting) issue in windows. Rmapi is working fine in windows !

You should open an issue on the rmapi repository, and attach the detailed log to the issue.

Run rmapi as follows:

```sh
$ RMAPI_TRACE=1 rmapi 2>&1 rmapi_log.txt
```
